### PR TITLE
[ENTSWM-540] new modul jax-rs 2.1

### PR DESCRIPTION
**Pull request by mirostary**

### CODE REVIEW

File: javaee8/jaxrs-2.1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>io.reactivex.rxjava2</groupId>
+            <artifactId>rxjava</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-rs-extension-rx2</artifactId>
+        </dependency>
```
Comment (Ladicek) [inline javaee8/jaxrs-2.1/pom.xml]: RESTEasy should support RxJava types itself. CXF is definitely a no-no.

Comment (Ladicek) [reply javaee8/jaxrs-2.1/pom.xml]: See https://docs.jboss.org/resteasy/docs/3.9.0.Final/userguide/html/Reactive.html

File: javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseClient.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return getClient("http://localhost:8080/broadcasting/register");
+    }
+
+    @GET
+    @Path("/broadcast")
+    public String getBroadcast() {
+        return getClient("http://localhost:8080/broadcasting/broadcast");
+    }
+
+    private static String getClient(String url) {
```
Comment (Ladicek) [inline javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseClient.java]: This method doesn't return a "client", so `getClient` is a bad name. I'd just call it `get`.

Comment (Ladicek) [reply javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseClient.java]: Or `getSse`, or something like that.

File: javaee8/jaxrs-2.1/src/test/java/org/wildfly/swarm/ts/javaee8/jaxrs/ReactiveClientTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package org.wildfly.swarm.ts.javaee8.jaxrs;
```
Comment (Ladicek) [inline javaee8/jaxrs-2.1/src/test/java/org/wildfly/swarm/ts/javaee8/jaxrs/ReactiveClientTest.java]: Should this test be in the `rclient` package?

File: javaee8/jaxrs-2.1/src/test/java/org/wildfly/swarm/ts/javaee8/jaxrs/SseTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package org.wildfly.swarm.ts.javaee8.jaxrs;
```
Comment (Ladicek) [inline javaee8/jaxrs-2.1/src/test/java/org/wildfly/swarm/ts/javaee8/jaxrs/SseTest.java]: Should this test be in the `sse` package?

Comment (Ladicek) [reply javaee8/jaxrs-2.1/src/test/java/org/wildfly/swarm/ts/javaee8/jaxrs/SseTest.java]: And use `@DefaultDeployment`.

File: pom.xml [hunk 5]
```diff
@@ -140,16 +141,18 @@
         <version.com.google.guava>25.0-jre</version.com.google.guava>
         <version.com.h2database.h2>1.4.200</version.com.h2database.h2>
         <version.com.icegreen.greenmail>1.5.11</version.com.icegreen.greenmail>
+        <version.cio.reactivex.rxjava2>2.2.15</version.cio.reactivex.rxjava2>
```
Comment (Ladicek) [inline pom.xml]: Typo, `cio` -> `io`.

File: pom.xml [hunk 6]
```diff
@@ -140,16 +141,18 @@
...
-        <version.io.thorntail>2.5.1.Final-SNAPSHOT</version.io.thorntail>
+        <version.io.thorntail>2.6.1.Final-SNAPSHOT</version.io.thorntail>
         <version.jakarta.enterprise.cdi-api>2.0.2</version.jakarta.enterprise.cdi-api>
         <version.jakarta.json-api>1.1.6</version.jakarta.json-api>
         <version.junit.junit>4.12</version.junit.junit>
         <version.net.java.xadisk>1.2.2</version.net.java.xadisk>
         <version.org.amqphub.jca.resource-adapter>1.0.0</version.org.amqphub.jca.resource-adapter>
         <version.org.apache.activemq.activemq-rar>5.15.10</version.org.apache.activemq.activemq-rar>
         <version.org.apache.activemq.examples.modules.artemis-rar>2.10.1</version.org.apache.activemq.examples.modules.artemis-rar>
+        <version.org.apache.cxf>3.3.4</version.org.apache.cxf>
```
Comment (Ladicek) [inline pom.xml]: As I said, this will need to go.

**Comment by Ladicek** [comment]
The general approach looks good, but some changes are needed.

File: pom.xml (hunk 6 above)
Comment (Ladicek) [reply pom.xml]: Please remove this.

**Comment by Ladicek** [comment]
Please remove the CXF stuff from `pom.xml`, otherwise LGTM.

File: javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseBroadcastResource.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    @GET
+    @Path("/broadcast")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void broadcast(@Context SseEventSink eventSink) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        broadcaster.register(eventSink);
+        latch.countDown();
+
+        if (!latch.await(10, TimeUnit.SECONDS)) {
```
Comment (Ladicek) [inline javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseBroadcastResource.java]: Actually, what is the latch doing here? I mean, what's the difference between

```java
        CountDownLatch latch = new CountDownLatch(1);
        broadcaster.register(eventSink);
        latch.countDown();
```

and simple

```java
        broadcaster.register(eventSink);
```

?

Comment (mirostary) [reply javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseBroadcastResource.java]: Right, It stayed there after I tested broadcast with more than one client. In this code, it is not needed. I will delete it.

**Comment by Ladicek** [comment]
I did a few little changes, but that was mostly cosmetics. Thanks!

**Pull request merged**